pkg/ruler/ruler.go

```diff
@@ -512,20 +524,41 @@ func (r *Ruler) run(ctx context.Context) error {
 func (r *Ruler) syncRules(ctx context.Context, reason string) {
 	level.Debug(r.logger).Log("msg", "syncing rules", "reason", reason)
 	r.rulerSync.WithLabelValues(reason).Inc()
+	timer := prometheus.NewTimer(nil)
+
+	defer func() {
+		ruleGroupSyncDuration := timer.ObserveDuration().Seconds()
+		r.ruleGroupSyncDuration.Set(ruleGroupSyncDuration)
```

Comment:
Yes this was intentional to include both loading time from storage and syncing time to Prometheus. I suspect the biggest chunk of time spent will be listing and loading rules from storage.  However, if you think otherwise, we can discuss and I'm open to changing

Reply:
I am ok with this. But it would be better to mention it in the metric name or metric help text.
Now it says `Total time taken to sync rule groups` so I thought it measures the duration of `r.manager.SyncRuleGroups(ctx, loadedConfigs)` call.

Reply:
Makes sense. I will change the help text and see if I can come up with a better metric name

Reply:
Changed the help text